### DISCUSSION TITLE
Problem with Chainlink Automation.

### DISCUSSION THREAD

**Discussion by EndixAK** [Cyfrin/foundry-full-course-cu#3023] [Q&A]
Hello! 
I have two questions, but before I ask them I would like to add that I have been looking for how to do this all day today but I have not succeeded.

1. a question regarding the Raffle Project: if we have a Chainlink VRF Subscription coded, is this automatically hooked up to Chainlink Automation that later on in the course Patrick no longer makes a subscription to Chainlink Automation?

2 Question regarding chainlink automation: does anyone know of a small project where upkeeps are done without a GUI like VRF for Raffle? I'm doing a FundMe project right now only expanded as a whole platform, it's already ready just need to do it without GUI.

Thank you in advance and best regards!

**Comment by EngrPips** [comment]
Hello @EndixAK

**ANSWER TO YOUR FIRST QUESTION**
Chainlink VRF and Automation are two different chainlink products, and I believe you need to create a subscription for `vrf` and you need to register a new upkeep for `automation.`

**ANSWER TO YOUR SECOND QUESTION**
I believe you can do chainlink automation without GUI. You need to get the address of the chainlink automation registry contract and call the necessary function on it with the appropriate calldata.

**Comment by KMean** [comment]
1. NO
2. I guess you're talking about registering and funding an upkeep via contract. Have a look at the [chainlink guide](https://docs.chain.link/chainlink-automation/guides/register-upkeep-in-contract) to do that. 
Alternatively check [this](https://ethereum.stackexchange.com/questions/131289/how-to-register-new-upkeep-on-chainlink-without-using-gui) which shows you one way of doing it [here](https://gist.github.com/stone4419/4b5129e184a20f6f85b2cc613381d54f). You probably will want to adapt it though.
